test/list_test.rb

```diff
@@ -576,6 +576,48 @@
     end
   end
 
+  describe '#each_node' do
+    it 'returns enumerator if no block given' do
+      assert_instance_of Enumerator, list.each
+    end
+
+    it 'pass each node data to the block' do
+      list.push(node_1)
+      list.push(node_2)
+      nodes = []
+      list.each_node { |e| nodes << e }
+      assert_equal %w(foo bar), nodes.map(&:data)
+    end
+  end
+
+  describe '#reverse_each' do
+    it 'returns enumerator if no block given' do
+      assert_instance_of Enumerator, list.each
+    end
+
+    it 'pass each node data to the block' do
+      list.push(node_1)
+      list.push(node_2)
+      nodes = []
+      list.reverse_each { |e| nodes << e }
+      assert_equal %w(bar foo), nodes
+    end
+  end
+
+  describe '#reverse_each_node' do
+    it 'returns enumerator if no block given' do
+      assert_instance_of Enumerator, list.each
+    end
+
+    it 'pass each node data to the block' do
+      list.push(node_1)
+      list.push(node_2)
+      nodes = []
+      list.reverse_each_node { |e| nodes << e }
+      assert_equal %w(bar foo), nodes.map(&:data)
+    end
+  end
+
```

Comment:
I missed in PR #3 , if it is not too much trouble, could you please add specs for `insert_before_node` and `insert_after_node` since they are part of public api.

Reply:
sure 👍